### PR TITLE
BUILD-87: Add support for build volumes

### PR DESCRIPTION
**Pull request by coreydaley**
Updating build types:
 - added listType to appropriate fields
 - added +required tag for required fields

### CODE REVIEW

File: build/v1/types.go [hunk 1]
```diff
@@ -746,6 +746,7 @@ type DockerBuildStrategy struct {
 	// volumes is a list of input volumes that can be mounted into the builds runtime environment.
 	// Only a subset of Kubernetes Volume sources are supported by builds.
 	// More info: https://kubernetes.io/docs/concepts/storage/volumes
+	// +listType=atomic
```
Comment (sttts) [inline build/v1/types.go:749]: these three markers contradict each-other

Comment (coreydaley) [reply build/v1/types.go]: updated in current push

**Comment by coreydaley** [comment]
@sttts ptal, comment has been addressed

File: build/v1/types.go [hunk 2]
```diff
@@ -1358,12 +1362,15 @@ type BuildVolume struct {
...
 	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+	// +required
 	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
 
 	// source represents the location and type of the mounted volume.
+	// +required
 	Source BuildVolumeSource `json:"source" protobuf:"bytes,2,opt,name=source"`
 
 	// mounts represents the location of the volume in the image build container
+	// +listType=atomic
```
Comment (sttts) [inline build/v1/types.go:1373]: just to be sure: you would expect that this is replaced completely instead of being merged?

Comment (coreydaley) [reply build/v1/types.go]: this is updated now, let me know if that makes sense.

**Pull request merged**